docs/_integration-with-existing-apps-ios.md

```diff
@@ -0,0 +1,605 @@
+<TabItem value="objc">
+
+```diff title="ReactViewController.m"
+#import "ReactViewController.h"
++#import <React-RCTAppDelegate/RCTRootViewFactory.h>
++#import <React-RCTAppDelegate/RCTAppDelegate.h>
+
+@interface ReactViewController ()
+
+@end
+
+@implementation ReactViewController
+
+ - (void)viewDidLoad {
+   [super viewDidLoad];
+   // Do any additional setup after loading the view.
++   RCTRootViewFactory *factory = ((RCTAppDelegate *)RCTSharedApplication().delegate).rootViewFactory;
++   self.view = [factory viewWithModuleName:@"HelloWorld"];
+ }
+
+@end
+```
+
+</TabItem>
+<TabItem value="swift">
+
+```diff title="ReactViewController.swift"
+import UIKit
++import React_RCTAppDelegate
+
+class ReactViewController: UIViewController {
+
+  override func viewDidLoad() {
+    super.viewDidLoad()
+
++    let factory = (RCTSharedApplication()?.delegate as? RCTAppDelegate)?.rootViewFactory
++    self.view = factory?.view(withModuleName: "HelloWorld")
+  }
+}
+```
+
+</TabItem>
+</Tabs>
+
+4. Make sure to disable the Sandbox scripting. To achieve this, in Xcode, click on your app, then on build settings. Filter for script and set the `User Script Sandboxing` to `NO`. This step is needed to properly switch between the Debug and Release version of the [Hermes engine](https://github.com/facebook/hermes/blob/main/README.md) that we ship with React Native.
+
+![Disable Sandboxing](/docs/assets/disable-sandboxing.png);
+
+## 6. Test your integration
+
+You have completed all the basic steps to integrate React Native with your application. Now we will start the [Metro bundler](https://metrobundler.dev/) to build your TypeScript application code into a bundle. Metro's HTTP server shares the bundle from `localhost` on your developer environment to a simulator or device. This allows for [hot reloading](https://reactnative.dev/blog/2016/03/24/introducing-hot-reloading).
+
+First, you need to create a `metro.config.js` file in the root of your project as follows:
+
+```js
+const {getDefaultConfig} = require('@react-native/metro-config');
+module.exports = getDefaultConfig(__dirname);
+```
+
+You can checkout the [metro.config.js file](https://github.com/react-native-community/template/blob/0.76-stable/template/metro.config.js) from the Community template file as reference.
+
+Once you have the config file in place, you can run the bundler. Run the following command in the root directory of your project:
+
+<Tabs groupId="package-manager" queryString defaultValue={constants.defaultPackageManager} values={constants.packageManagers}>
+<TabItem value="npm">
+
+```shell
+npm start
+```
+
+</TabItem>
+<TabItem value="yarn">
+
+```shell
+yarn start
+```
+
+</TabItem>
+</Tabs>
+
+Now build and run your iOS app as normal.
+
+Once you reach your React-powered Activity inside the app, it should load the JavaScript code from the development server and display:
+
+![Screenshot](/docs/assets/EmbeddedAppIOSVideo.gif)
```

Comment:
Can we constrain the width (see [example code here](https://github.com/facebook/react-native-website/blob/61313ad2e9f756ea0c3e6dd87a1fd53e1f674d6e/docs/datepickerios.md?plain=1#L90))?

Currently it's like this:
![CleanShot 2024-10-02 at 12 05 52@2x](https://github.com/user-attachments/assets/91ebdb9a-0e02-42b1-abff-3da2c04d6845)

If I set a max with of 300px it looks like this:
![CleanShot 2024-10-02 at 12 06 28@2x](https://github.com/user-attachments/assets/3b65df8c-9ab3-42cb-a875-ab486a3b491c)

Reply:
we need to do it on android as well.